sniffles/regex_generator.py

```diff
@@ -25,78 +25,140 @@
-        elif opt == "-g":
-            groups = True
-        elif opt == "-l":
-            lambd = int(arg)
-            if lambd <= 0:
-                lambd = 10
-        elif opt == "-M":
-            if arg is not None:
-                max_regex_length = int(arg)
-        elif opt == "-m":
-            min_regex_length = int(arg)
-            if min_regex_length < 1:
-                min_regex_length = 1
-        elif opt == "-n":
-            negation_prob = int(arg)
-            if negation_prob < 0:
-                negation_prob = 0
-        elif opt == "-o":
-            option_chance = int(arg)
-            if option_chance < 0:
-                option_chance = 0
-        elif opt == "-R":
-            rep_chance = int(arg)
-            if rep_chance < 0:
-                rep_chance = 0
-        elif opt == "-r":
-            if arg is not None:
-                rep_dist = re.split('[\s,;]+', arg)
-        elif opt == "-t":
-            if arg is not None:
-                type_dist = re.split('[\s,;]+', arg)
-        elif opt == "-?":
-            usage()
-        else:
-            print("Unrecognized Option: ", opt)
-            usage()
-    create_regex_list(number, lambd, type_dist, char_dist, class_dist,
-                      rep_dist, rep_chance, option_chance, negation_prob,
-                      re_file, min_regex_length, max_regex_length, groups)
+
+    parser = argparse.ArgumentParser(description='''
+    Random Regular Expression Generator
+    will create random regular expressions.  It is possible
+    to tune the structures within the regular expressions to a prbability
+    distribution, but currently not the content.  This is desirable in
+    order to explore the maximum diversity in possible regular expressions
+    (though not necessarily realistic regular expressions).
+    The distributions are handled by creating a list of probabilities for
+    the various possibilities, or slots, for a particular distribution.
+    These are added as command line arguments using a simple string
+    list like: "10,30,40,20".  The list should have as many values
+    as it has slots.  The total of all values in the list should be
+    100 and there should not be any fractions.  The value at each slot
+    is the probability that that slot will be chosen.  For example,
+    the base RE structural type distribution has three slots.  The
+    first slot is the probability that the next structure type is
+    a character (where a character can be a letter, digit, binary, ASCII,
+    or substitution class (like \\w).  The second slot is for character
+    classes like [ab@%], [^123], or [a-z].  The final slot is the probability
+    of alternation occuring like (ab|cd).  With these three slots you can tune
+    how often you would like the structures to appear in your regular
+    expressions.  For example, python3.4 -c 10 -t "80,10,10" would create
+    10 regular expressions where 80% of the structures used would be
```

Comment:
I noticed below we have %% for percent, but here only one.  Is that significant?

Reply:
Yes. For some reason, `description` is treated as a normal string while `help` is treated as a format string, where `%` has a special meaning.